cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Node/LedgerEvents/Governance/ProposeNewConstitution.hs

```diff
@@ -0,0 +1,414 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE NumericUnderscores #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Cardano.Testnet.Test.Node.LedgerEvents.Governance.ProposeNewConstitution
+  ( hprop_ledger_events_propose_new_constitution
+  ) where
+
+import           Cardano.Api
+import           Cardano.Api.Shelley
+
+import           Cardano.Testnet
+
+import           Prelude
+
+import           Cardano.Crypto.Hash.Class
+import qualified Cardano.Ledger.Conway.Governance as Ledger
+import qualified Cardano.Ledger.SafeHash as Ledger
+import           Control.Monad
+import           Control.Monad.Trans.Except
+import           Control.Monad.Trans.Except.Extra
+import qualified Data.Map.Strict as Map
+import           Data.String
+import qualified Data.Text as Text
+import           Data.Word
+import           GHC.IO.Exception (IOException)
+import           GHC.Stack (HasCallStack, callStack)
+import           Lens.Micro
+import           System.FilePath ((</>))
+
+import           Hedgehog
+import qualified Hedgehog.Extras as H
+import qualified Hedgehog.Extras.Stock.IO.Network.Sprocket as IO
+import qualified Testnet.Process.Cli as P
+import qualified Testnet.Process.Run as H
+
+import           Testnet.Components.SPO
+import qualified Testnet.Property.Utils as H
+import           Testnet.Runtime
+
+
+newtype AdditionalCatcher
+  = IOE IOException
+  deriving Show
+
+
+hprop_ledger_events_propose_new_constitution :: Property
+hprop_ledger_events_propose_new_constitution = H.integrationRetryWorkspace 2 "ledger-events-governance-propose-new-constitution" $ \tempAbsBasePath' -> do
+  -- Start a local test net
+  conf@Conf { tempAbsPath } <- H.noteShowM $ mkConf tempAbsBasePath'
+  let tempAbsPath' = unTmpAbsPath tempAbsPath
+      tempBaseAbsPath = makeTmpBaseAbsPath tempAbsPath
+
+  work <- H.createDirectoryIfMissing $ tempAbsPath' </> "work"
+
+  let sbe = ShelleyBasedEraConway
+      era = toCardanoEra sbe
+      cEra = AnyCardanoEra era
+      fastTestnetOptions = cardanoDefaultTestnetOptions
+        { cardanoEpochLength = 100
+        , cardanoSlotLength = 0.1
+        , cardanoNodeEra = cEra
+        }
+
+  testnetRuntime@TestnetRuntime
+    { testnetMagic
+    , poolNodes
+    , wallets
+    }
+    <- cardanoTestnet fastTestnetOptions conf
+
+  poolNode1 <- H.headM poolNodes
+
+  poolSprocket1 <- H.noteShow $ nodeSprocket $ poolRuntime poolNode1
+
+  execConfig <- H.mkExecConfig tempBaseAbsPath poolSprocket1
+
+  let socketName' = IO.sprocketName poolSprocket1
+      socketBase = IO.sprocketBase poolSprocket1 -- /tmp
```

Comment:
What's the significance of the `/tmp` comment?

Reply:
Just to remind myself this is the base dir.